src/Tasks.UnitTests/CSharpParserUtilitites_Tests.cs

```diff
@@ -41,58 +48,76 @@ public void NoNamespace()
             AssertParse("class MyClass{}", "MyClass");
         }
 
-        [Fact]
-        public void SneakyComment()
+        [Theory]
+        [InlineData("/*namespace MyNamespace { */ class MyClass {} /* } */", "MyClass")]
+        [InlineData("/*namespace MyNamespace ; */ class MyClass {}", "MyClass")] // file-scoped namespaces
+        public void SneakyComment(string fileContents, string expected)
         {
-            AssertParse("/*namespace MyNamespace { */ class MyClass {} /* } */", "MyClass");
+            AssertParse(fileContents, expected);
         }
 
-        [Fact]
-        public void CompoundNamespace()
+        [Theory]
+        [InlineData("namespace MyNamespace.Feline { class MyClass {} }", "MyNamespace.Feline.MyClass")]
+        [InlineData("namespace MyNamespace.Feline ; class MyClass {} ", "MyNamespace.Feline.MyClass")] // file-scoped namespaces
+        public void CompoundNamespace(string fileContents, string expected)
         {
-            AssertParse("namespace MyNamespace.Feline { class MyClass {} }", "MyNamespace.Feline.MyClass");
+            AssertParse(fileContents, expected);
         }
 
-        [Fact]
-        public void NestedNamespace()
+        [Theory]
+        [InlineData("namespace MyNamespace{ namespace Feline {class MyClass {} }}", "MyNamespace.Feline.MyClass")]
+        [InlineData("namespace MyNamespace; namespace Feline ;class MyClass {} ", "MyNamespace.Feline.MyClass")] // file-scoped namespaces
+        public void NestedNamespace(string fileContents, string expected)
         {
-            AssertParse("namespace MyNamespace{ namespace Feline {class MyClass {} }}", "MyNamespace.Feline.MyClass");
+            AssertParse(fileContents, expected);
         }
 
-        [Fact]
-        public void NestedNamespace2()
+        [Theory]
+        [InlineData("namespace MyNamespace{ namespace Feline {namespace Bovine{public sealed class MyClass {} }} }", "MyNamespace.Feline.Bovine.MyClass")]
+        [InlineData("namespace MyNamespace; namespace Feline ;namespace Bovine;public sealed class MyClass {}", "MyNamespace.Feline.Bovine.MyClass")] // file-scoped namespaces
```

Comment:
Since this is invalid C# I'm not sure we should have a test for it--if we change the implementation so that it overwrites (say) on a second file-scoped namespace that shouldn't break any users, so why fail the test?

Reply:
(and similar for all the cases below)

Reply:
I'll revert theory-ified tests that were modified to account for a broken file-scoped namespace test.